src/components/modal/addtostudio/container.jsx

```diff
@@ -0,0 +1,72 @@
+const bindAll = require('lodash.bindall');
+const PropTypes = require('prop-types');
+const React = require('react');
+const AddToStudioModalPresentation = require('./presentation.jsx');
+
+class AddToStudioModal extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleRequestClose',
+            'handleSubmit'
+        ]);
+
+        this.state = {
+            waitingToClose: false
+        };
+    }
+
+    componentWillUpdate () {
+        this.checkIfFinishedUpdating();
+    }
+
+    hasOutstandingUpdates () {
+        return (this.props.studios.some(studio => (studio.hasRequestOutstanding === true)));
+    }
+
+    checkIfFinishedUpdating () {
```

Comment:
This name is slightly confusing in the context of `componentWillUpdate`, since it doesn't look like it does anything. I think a better name would be `closeIfFinishedUpdating`.

Reply:
Agreed.